### PR TITLE
Add tests for CASSANDRA-10134

### PR DESCRIPTION
**Pull request by beobal**

### CODE REVIEW

File: replace_address_test.py [hunk 1]
```diff
@@ -225,6 +228,119 @@ def replace_first_boot_test(self):
...
+        cluster.seeds.remove(node3)
+        NUM_TOKENS = os.environ.get('NUM_TOKENS', '256')
+        if DISABLE_VNODES:
+            cluster.set_configuration_options(values={'initial_token': None, 'num_tokens': 1})
+        else:
+            cluster.set_configuration_options(values={'initial_token': None, 'num_tokens': NUM_TOKENS})
+        cluster.start()
+
+        debug("Inserting Data...")
+        if cluster.version() < "2.1":
```
Comment (ptnapoleon) [inline replace_address_test.py]: We don't test before 2.1, this check isn't needed.

File: replace_address_test.py [hunk 2]
```diff
@@ -225,6 +228,119 @@ def replace_first_boot_test(self):
...
+            node1.stress(['-o', 'insert', '--num-keys=10000', '--replication-factor=3'])
+        else:
+            node1.stress(['write', 'n=10000', '-schema', 'replication(factor=3)'])
```
Comment (ptnapoleon) [inline replace_address_test.py]: `n=10k` please

Comment (beobal) [inline replace_address_test.py]: fixed in both new tests

File: replace_address_test.py (hunk 1 above)
Comment (beobal) [inline replace_address_test.py]: good point, well made

File: replace_address_test.py [hunk 3]
```diff
@@ -225,6 +228,113 @@ def replace_first_boot_test(self):
...
+        replace a node without bootstrapping. In such scenarios best practice
+        advice has been to wipe the node's system keyspace data, set the initial
+        tokens via cassandra.yaml, startup without bootstrap and then repair.
+        Starting the node as a replacement allows the tokens to be learned from
+        gossip, but previously required auto_bootstrap=true. Since CASSANDRA-10134
+        replacement is allowed without bootstrapping, but it requires the operator
+        to acknowledge the risk in doing so by setting the cassandra.allow_unsafe_replace
+        system property at startup.
+
+        @jira_ticket=CASSANDRA-10134
```
Comment (ptnapoleon) [inline replace_address_test.py]: Can you have a space here, not an `=`, the way you do above?

File: replace_address_test.py [hunk 4]
```diff
@@ -225,6 +228,113 @@ def replace_first_boot_test(self):
...
+            debug("Starting node 3 with auto_bootstrap = {val}".format(val=auto_bootstrap))
+            node3.start(wait_other_notice=False)
+            node3.watch_log_for('Use cassandra.replace_address if you want to replace this node', from_mark=mark, timeout=20)
+            mark = node3.mark_log()
+
+    @since('3.6')
+    def unsafe_replace_test(self):
+        """
+        To handle situations such as failed disk in a JBOD, it may be desirable
+        replace a node without bootstrapping. In such scenarios best practice
```
Comment (ptnapoleon) [inline replace_address_test.py]: nit: "desirable `to` replace a node"

**Comment by ptnapoleon** [comment]
+1 after nits. I'm not really sure what Travis is upset about. I think it's safe to ignore.

**Comment by beobal** [comment]
Thanks, I've fixed the nits, rebased, squashed and merged.

**Pull request closed without merging**